Pokedex - Module 2 - Group Capstone Project

JuanDiegoGuerra opened the pull request:
### JavaScript capstone project - Your API-based webapp ( Module 2 )
 - Followed the project requirements and completed the 3 days Milestones of **"Your API-based webapp"** we completed the main requirements:
   - A home page showing a list of items that you can "like." ✅
   - A popup window with more data about an item that you can use to comment on it. ✅
   - Professional Documentation in a Readme file. ✅
- As part of the requirements, we recorded a video explaining how the page works. 
   - You can watch it here: [Loom Video - Pokedex - Module 2 - Group Capstone](https://www.loom.com/share/1446ee58f4574b658372ef24167a1272) ✅

**Preview : 🕵🏼‍♂️**

![final_preview_pokedex](https://github.com/JuanDiegoGuerra/Module_2-Capstone/assets/45726631/50bc3a41-45f7-4fb6-adb3-389d1cf8b2ca)

lincoln1883 committed:
Add linter and webpack setup config files.

lincoln1883 committed:
Add initial build for the project.

lincoln1883 committed:
Create files with minimal code.

lincoln1883 committed:
Add image for the logo.

lincoln1883 committed:
Add style rule for the header.

lincoln1883 committed:
Add containers for the logo and nav links.

lincoln1883 committed:
Add import for the header module.

lincoln1883 committed:
Create module folder and file to separate concerns

lincoln1883 committed:
Add latest build folder.

lincoln1883 committed:
Fix linter errors.

lincoln1883 committed:
Fix linter error.

lincoln1883 committed:
Run linter and build dist.

lincoln1883 committed:
To fix lighthouse error i added meaning less code.

lincoln1883 committed:
Merge pull request #23 from JuanDiegoGuerra/feature-page-header

Add header - group task #17

lincoln1883 committed:
Merge branch 'development' into feature-project-setup

lincoln1883 committed:
Merge pull request #22 from JuanDiegoGuerra/feature-project-setup

Set up the repository and tools.

JuanDiegoGuerra committed:
Media Files for "PokeCards"

JuanDiegoGuerra committed:
Update index.html

JuanDiegoGuerra committed:
Create pokemons.js

JuanDiegoGuerra committed:
Create api_data.js

JuanDiegoGuerra committed:
Create likes.js

JuanDiegoGuerra committed:
Update style.css

JuanDiegoGuerra committed:
Update index.js

JuanDiegoGuerra committed:
Autogenerated code

npm run build

JuanDiegoGuerra committed:
Create settings.json

JuanDiegoGuerra committed:
Fix Linters

Fixed: Eslint & Stylelint linters errors

JuanDiegoGuerra committed:
Add Pointer Cursor Style

JuanDiegoGuerra committed:
Update api_data.js

Using a new api token

JuanDiegoGuerra committed:
Required Changes

Using main api file to import api URLs to other modules

JuanDiegoGuerra committed:
Update index.js

npm run build

JuanDiegoGuerra committed:
Fix Linters

Fixed Eslint linter erros

lincoln1883 committed:
Merge pull request #24 from JuanDiegoGuerra/render-data

Displaying Pokemon Data from API

lincoln1883 committed:
Add api call and removed hard coded count.

lincoln1883 committed:
Removed comments and moved api call to header file

lincoln1883 committed:
Change card size so the fit better on the page.

JuanDiegoGuerra committed:
Add Media Files

2 svg files
1 jpg file

JuanDiegoGuerra committed:
Update header.js

Changed Header Logo img

JuanDiegoGuerra committed:
Update pokemons.js

Removed "reservation" button since it won't be implemented for this project

Co-Authored-By: Lincoln Gibson <lincolngibson7@gmail.com>

JuanDiegoGuerra committed:
Update style.css

Added styles for main page, cards, header, etc.

Co-Authored-By: Lincoln Gibson <lincolngibson7@gmail.com>

JuanDiegoGuerra committed:
Auto-generated files

npm run build

Co-Authored-By: Lincoln Gibson <lincolngibson7@gmail.com>

lincoln1883 committed:
Add hard-coded values with popup feature.

lincoln1883 committed:
Add styling for the popup.

lincoln1883 committed:
Imported the popup  and connect it to the card.

lincoln1883 committed:
Add comments module.

JuanDiegoGuerra committed:
Fix Linters

lincoln1883 committed:
Fix linter errors.

lincoln1883 committed:
Merge pull request #26 from JuanDiegoGuerra/feature-comment-popup

Display Comments popup

JuanDiegoGuerra committed:
Merge branch 'development' into Styles

JuanDiegoGuerra committed:
Merge pull request #25 from JuanDiegoGuerra/Styles

Main Styles of the project

lincoln1883 committed:
Add logic to function to display specific item.

lincoln1883 committed:
Add logic to ensure id is provided to the event.

lincoln1883 committed:
Edit code to render all items to prevent errors.

lincoln1883 committed:
Removed this unused file.

lincoln1883 committed:
Add import for api to send and receive data.

lincoln1883 committed:
Created api control center for comments.

lincoln1883 committed:
Remove redundant styles and adjust pop display.

lincoln1883 committed:
Add final build.

JuanDiegoGuerra committed:
Update index.html

Created Footer Section

JuanDiegoGuerra committed:
Update style.css

Added styles for new Footer section

JuanDiegoGuerra committed:
Autogenerated files

npm run build

Co-Authored-By: Lincoln Gibson <lincolngibson7@gmail.com>

JuanDiegoGuerra committed:
Create pokedex_capstone_preview.JPG

Media File for Readme Logo

JuanDiegoGuerra committed:
Update README.md

Added all the content for the readme file

lincoln1883 committed:
Add more details for the pokemon.

lincoln1883 committed:
Create counter the comments.

- Add counter module for comments and passed the comments object as argument. :+1:

lincoln1883 committed:
Modified code to account for correct test output.

lincoln1883 committed:
Modified code to get the correct return Data type.

lincoln1883 committed:
Create test folder,file and function for comments

lincoln1883 committed:
Add new edge case to the test.

lincoln1883 committed:
Merge pull request #27 from JuanDiegoGuerra/feature-get-comment-data

Display comments for a given item on the Comments pop-up

lincoln1883 committed:
Merge pull request #28 from JuanDiegoGuerra/featute-comment-counter

Add comment counter

lincoln1883 committed:
Merge pull request #29 from JuanDiegoGuerra/comments-counter-test

Comments counter testing

JuanDiegoGuerra committed:
Create pokemons.js

Mock structure to use it for testing

JuanDiegoGuerra committed:
Create pokemonsCounter.test.js

Pokemons Home Page counter Test
3 tests made

JuanDiegoGuerra committed:
Update index.html

Deleting an unnecesary line

JuanDiegoGuerra committed:
Fix Linters

Eslint linter and Stylelint linter errors

JuanDiegoGuerra committed:
Fix Linter

Stylint linter error

JuanDiegoGuerra committed:
Add new Dependency

JuanDiegoGuerra committed:
Fix Linters

Eslint Linters fix

JuanDiegoGuerra committed:
Autogenerated file

npx run build

Co-Authored-By: Lincoln Gibson <lincolngibson7@gmail.com>

JuanDiegoGuerra committed:
Merge pull request #30 from JuanDiegoGuerra/Footer

Footer Section

JuanDiegoGuerra committed:
Merge pull request #31 from JuanDiegoGuerra/Readme

Readme - Documentation

JuanDiegoGuerra committed:
Merge branch 'development' into Testing-PokemonsCount

JuanDiegoGuerra committed:
Merge pull request #32 from JuanDiegoGuerra/Testing-PokemonsCount

Testing pokemons count

JuanDiegoGuerra committed:
Create new Unit Test for Pokemon Counter

lincoln1883 committed:
Update readme.md

Added more in details to the readme to comply with the requirements.U

JuanDiegoGuerra committed:
Merge pull request #33 from JuanDiegoGuerra/Testing-PokemonsCount

Create new Unit Test for Pokemon Counter

lincoln1883 committed:
Update README.md

Add update to the social media section.

lincoln1883 committed:
Merge pull request #34 from JuanDiegoGuerra/edit-readme-file

Update readme.md

Lucas-Masaba commented on src/modules/pokemons.js:
```diff
@@ -0,0 +1,53 @@
+            </div> 
+            <p class="like-count" id="like-count${id + 1}">0 Likes</p>
+            <div class="buttons">
+              <button type="button" id="${id + 1}" class="comment">Comments</button>
```
- [ ] I am able to see the comments only on the mobile view. I cannot see them on the desktop view. Even on the mobile view, I find it hard to read the comments. Please ensure that the comments written can be seen clearly as seen in the wireframe :+1: 

### View of comments on the mobile screen
![image](https://github.com/JuanDiegoGuerra/Module_2-Capstone/assets/45460192/fe550cc5-893a-466c-802a-f54f3b35ce6d)

### View of comments on the desktop view
![image](https://github.com/JuanDiegoGuerra/Module_2-Capstone/assets/45460192/ad2a8e24-b6ae-4e11-8cf0-4a334973a9ef)

Lucas-Masaba reviewed (changes requested):
# Hello @JuanDiegoGuerra  and @lincoln1883  👋 , 

Good job so far! 👍 
There are some issues that you still need to work on to go to the next project but you are almost there!

![almost-there](https://media.giphy.com/media/ReyD4OdRiHeLDCRaEP/giphy.gif)

## Highlights 🍾 
- Linter checks and unit tests are passing :heavy_check_mark: 
- Good use of ES6 :surfer: 
- Well-written commit messages :+1: 
- Successfully used an API :snowboarder: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions 💭 
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @Lucas-Masaba in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

JuanDiegoGuerra committed:
Update popUp.js

Add a class to an h3 element

Co-Authored-By: Lincoln Gibson <lincolngibson7@gmail.com>

JuanDiegoGuerra committed:
Update style.css

Added Media Query Styles

Co-Authored-By: Lincoln Gibson <lincolngibson7@gmail.com>

JuanDiegoGuerra committed:
Requested Changes

Fixed styles for popup window.

Co-Authored-By: Lincoln Gibson <lincolngibson7@gmail.com>

JuanDiegoGuerra committed:
Add Limit Chars to TextArea

Text Area input to add a comment

Co-Authored-By: Lincoln Gibson <lincolngibson7@gmail.com>

JuanDiegoGuerra commented:
###  Required changes made by @JuanDiegoGuerra & @lincoln1883  🛠
**All linters are enable!**

- Added Media Query ✅
- Fixed _"I am able to see the comments only on the mobile view. I cannot see them on the desktop view. Even on the mobile view, I find it hard to read the comments. Please ensure that the comments written can be seen clearly as seen in the wireframe 👍"_ ✅
  - Now the app is working correctly!

![todolist1](https://user-images.githubusercontent.com/45726631/231703283-083f07c2-f037-4fa4-a834-be7d3bd26e06.gif)

**Preview 🕵🏼‍♂️:**

![requested_changes_preview](https://github.com/JuanDiegoGuerra/Module_2-Capstone/assets/45726631/72eeed69-7dc3-4dbc-82b4-c6c16957f5d5)

jerryowusu reviewed (approved):
Hi @JuanDiegoGuerra and @lincoln1883 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


![weldone](https://github.com/JuanDiegoGuerra/Module_2-Capstone/assets/65233860/925b7eea-f1b4-46c5-92d0-f0dcc0f77b7e)


## Highlights
Nice design :+1: 
API working fine :+1: 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
